### PR TITLE
add sushiswap binance and polygon

### PR DESCRIPTION
**Pull request by developerfred**
Brief comments on the purpose of your changes:
```
• Added Sushiswap dex trades on Binance to dex.trades table.
• Added Sushiswap dex trades on Polygon to dex.trades table.
• Modified the dex.trades model
- Updated dbt_project.yml file
```

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [x] `coin_id` represents the ID of the coin on coinpaprika.com
* [x] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: seeds/dex/trades/dex_trades_seed.csv [hunk 1]
```diff
@@ -123,4 +123,16 @@ bnb,biswap,1,2022-11-22,0x66a602998a3585037eb498c103b009a4face2faf93ea57dfb1a44d
...
 bnb,biswap,1,2022-11-22,0xe2efd633720bd29322da7e39857e98b3c9904a1c55f9b3995441160983e06296,119,0xe9e7cea3dedca5984780bafc599bd69add087d56,43.898605549510836795,0x563ca064e41f3b5d80adeecfe49ab375fd7afbef,2756.0281
 bnb,biswap,1,2022-11-22,0x0b9ba3ee4a0b3f4db2baba70be527f5a50aea77423cd921106ec9f0ac8dcf169,115,0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c,0.018248512111984361,0xe9e7cea3dedca5984780bafc599bd69add087d56,4.837104443141354752 
-bnb,biswap,1,2022-11-22,0xaf03946d9b98e5d6e841bb50555208801e48d773a8dfdf1ecc203d943c27c603,350,0x2170ed0880ac9a755fd29b2688956bd959f933f8,6.247796965743473757,0x7130d2a12b9bcbfae4f2634d864a1ee1ce3ead9c,0.43849
\ No newline at end of file
+bnb,biswap,1,2022-11-22,0xaf03946d9b98e5d6e841bb50555208801e48d773a8dfdf1ecc203d943c27c603,350,0x2170ed0880ac9a755fd29b2688956bd959f933f8,6.247796965743473757,0x7130d2a12b9bcbfae4f2634d864a1ee1ce3ead9c,0.43849
+  bnb,sushiswap,1,2022-11-22,0x683b4299466b4fec3fac2859e2dd27215f69879b31a7384593b46aef71952ed5,373,0xa0ed3c520dc0632657ad2eaaf19e26c4fd431a84,6,571.818499746365907562,0xe9e7cea3dedca5984780bafc599bd69add087d56,20
```
Comment (Hosuke) [inline seeds/dex/trades/dex_trades_seed.csv:127]: This line caused the dbt seed error.

Comment (Hosuke) [reply seeds/dex/trades/dex_trades_seed.csv:127]: especially `6,571.818499746365907562` this part.

**Comment by Hosuke** [comment]
Hi @developerfred , the table `sushiswap_polygon.UniswapV2Pair_evt_Swap` is currently unavailable in Dune V2 engine. Do you think `sushi_polygon.UniswapV2Pair_evt_Swap` is suitable for the replacement?

**Comment by developerfred** [comment]
> Hi @developerfred , the table `sushiswap_polygon.UniswapV2Pair_evt_Swap` is currently unavailable in Dune V2 engine. Do you think `sushi_polygon.UniswapV2Pair_evt_Swap` is suitable for the replacement?

done [7dbdd3d](https://github.com/duneanalytics/spellbook/pull/2000/commits/7dbdd3d88d33358b2ef4027f70d4ccbef59a3b6f)

File: models/sushiswap/optimism/sushi_optimism_trades.sql [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            ''                                                           AS trace_address,
+            t.evt_index
+    FROM {{ source('sushi_optimism', 'ConstantProductPool_evt_Swap') }} t
+    INNER JOIN {{ source('sushi_optimism', 'UniswapV2Factory_evt_PairCreated') }} f
+        ON f.pair = t.contract_address
+    {% if is_incremental() %}
+    WHERE t.evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% else %}
+    WHERE t.evt_block_time >= '{{ project_start_date }}'
+    {% endif %}
```
Comment (Hosuke) [inline models/sushiswap/optimism/sushi_optimism_trades.sql:37]: `sushi_optimism_trades` is somehow different from sushiswap on other chains. The approach using `sushi_optimism.UniswapV2Factory_evt_PairCreated` may not work in this situation. You can try another approach, using `sushi_optimism.StablePool_evt_Swap`, and find out the pools' information from other factory contract.

**Comment by developerfred** [comment]
> If `sushi_optimism_trades` is hard to implement at the moment, we can remove the optimism_trades from this PR, and review and merge other models for sushiswap first.

Good idea, I’ll open a new pr only with the optimism, thank you.

File: models/sushiswap/optimism/sushi_optimism_trades.sql (hunk 2 above)
Comment (developerfred) [reply models/sushiswap/optimism/sushi_optimism_trades.sql]: I'll create a new pull request using this tip, thanks.

**Comment by Hosuke** [comment]
@developerfred I will try to make this PR ready-for-review in the next few commits.

File: seeds/dex/trades/dex_trades_seed.csv [hunk 3]
```diff
@@ -146,6 +146,15 @@ bnb,biswap,1,2022-11-22,0x9050a1e2f7aa392e39195b5b22382d595e94104b72699655b198c4
...
+polygon,sushiswap,1,2022-11-22,0xcd4fda9b8492b26fb927a37f2276066d4e845d5831898c6aa3bc52195e64be29,51,0x4c3bf0a3de9524af68327d1d2558a3b70d17d42a,10,0x2791bca1f2de4661ed88a30c99a7a9449aa84174,17.471735
+polygon,sushiswap,1,2022-11-22,0xf993683656e9d580099b359bf697a15d2c704bac55a5341df59d266e00bd2214,95,0x2b9e7ccdf0f4e5b24757c1e1a80e311e34cb10c7,10,0x2791bca1f2de4661ed88a30c99a7a9449aa84174,23.507209
+polygon,sushiswap,1,2022-11-22,0x581a3ed5905f52623813259451c9bba0fb4a1b902bd2c84a04bf26831384221c,33,0xd125443f38a69d776177c2b9c041f462936f8218,160.621170255560769709,0x7ceb23fd6bc0add59e62ac25578270cff1b9f619,0.041137
+polygon,sushiswap,1,2022-11-22,0xb3fd8d3ea108e9bb55221d12e9c3d8a315e5eb1c7aad884817cdfe1d10ac0d35,38,0xd125443f38a69d776177c2b9c041f462936f8218,61.238323253823349251,0x960d43be128585ca45365cd74a7773b9d814dfbe,33.502071
+polygon,sushiswap,1,2022-11-22,0x65c15eccd09b70a984d39c61acd4d3e7b5dfa461ac761ccfdc87bae92077391f,42,0xd125443f38a69d776177c2b9c041f462936f8218,26.343507111310067208,0x60ed6acef3a96f8cdaf0c0d207bbafa66e751af2,0.087984
```
Comment (Hosuke) [inline seeds/dex/trades/dex_trades_seed.csv:157]: <table role="table" class="table_table__fuS_N" style="margin: 0px; padding: 0px; box-sizing: inherit; width: 1382.43px; border-collapse: collapse; font-family: &quot;IBM Plex Mono&quot;; font-size: 1.2rem;"><thead style="margin: 0px; padding: 0px; box-sizing: inherit;"><tr role="row" style="margin: 0px; padding: 0px; box-sizing: inherit;"><th colspan="1" role="columnheader" aria-sort="none" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; text-align: inherit; font-weight: 500; position: sticky; top: 0px; height: 3rem; white-space: nowrap; background: var(--color1-200);"><button type="button" title="Toggle SortBy" style="margin: 0px; padding: 0px; box-sizing: inherit; font-weight: inherit; cursor: pointer; text-align: inherit; font-size: inherit; font-family: inherit; line-height: inherit; background: transparent; border: none; color: inherit; display: block; width: 151.203px; user-select: text;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;">test_description</div></button></th><th colspan="1" role="columnheader" aria-sort="none" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; text-align: inherit; font-weight: 500; position: sticky; top: 0px; height: 3rem; white-space: nowrap; background: var(--color1-200);"><button type="button" title="Toggle SortBy" style="margin: 0px; padding: 0px; box-sizing: inherit; font-weight: inherit; cursor: pointer; text-align: inherit; font-size: inherit; font-family: inherit; line-height: inherit; background: transparent; border: none; color: inherit; display: block; width: 100.805px; user-select: text;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;">result (model)</div></button></th><th colspan="1" role="columnheader" aria-sort="none" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; text-align: inherit; font-weight: 500; position: sticky; top: 0px; height: 3rem; white-space: nowrap; background: var(--color1-200);"><button type="button" title="Toggle SortBy" style="margin: 0px; padding: 0px; box-sizing: inherit; font-weight: inherit; cursor: pointer; text-align: inherit; font-size: inherit; font-family: inherit; line-height: inherit; background: transparent; border: none; color: inherit; display: block; width: 108px; user-select: text;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;">expected (seed)</div></button></th><th colspan="1" role="columnheader" aria-sort="none" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; text-align: inherit; font-weight: 500; position: sticky; top: 0px; height: 3rem; white-space: nowrap; background: var(--color1-200);"><button type="button" title="Toggle SortBy" style="margin: 0px; padding: 0px; box-sizing: inherit; font-weight: inherit; cursor: pointer; text-align: inherit; font-size: inherit; font-family: inherit; line-height: inherit; background: transparent; border: none; color: inherit; display: block; width: 115.203px; user-select: text;"><div class="visual_datetime__8Dkmt" style="margin: 0px; padding: 0px; box-sizing: inherit; white-space: nowrap;">block_date</div></button></th><th colspan="1" role="columnheader" aria-sort="none" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; text-align: inherit; font-weight: 500; position: sticky; top: 0px; height: 3rem; white-space: nowrap; background: var(--color1-200);"><button type="button" title="Toggle SortBy" style="margin: 0px; padding: 0px; box-sizing: inherit; font-weight: inherit; cursor: pointer; text-align: inherit; font-size: inherit; font-family: inherit; line-height: inherit; background: transparent; border: none; color: inherit; display: block; width: 72px; user-select: text;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;">blockchain</div></button></th><th colspan="1" role="columnheader" aria-sort="none" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; text-align: inherit; font-weight: 500; position: sticky; top: 0px; height: 3rem; white-space: nowrap; background: var(--color1-200);"><button type="button" title="Toggle SortBy" style="margin: 0px; padding: 0px; box-sizing: inherit; font-weight: inherit; cursor: pointer; text-align: inherit; font-size: inherit; font-family: inherit; line-height: inherit; background: transparent; border: none; color: inherit; display: block; width: 64.8047px; user-select: text;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;">project</div></button></th><th colspan="1" role="columnheader" aria-sort="none" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; text-align: inherit; font-weight: 500; position: sticky; top: 0px; height: 3rem; white-space: nowrap; background: var(--color1-200);"><button type="button" title="Toggle SortBy" style="margin: 0px; padding: 0px; box-sizing: inherit; font-weight: inherit; cursor: pointer; text-align: inherit; font-size: inherit; font-family: inherit; line-height: inherit; background: transparent; border: none; color: inherit; display: block; width: 50.4062px; user-select: text;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;">version</div></button></th><th colspan="1" role="columnheader" aria-sort="none" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; text-align: inherit; font-weight: 500; position: sticky; top: 0px; height: 3rem; white-space: nowrap; background: var(--color1-200);"><button type="button" title="Toggle SortBy" style="margin: 0px; padding: 0px; box-sizing: inherit; font-weight: inherit; cursor: pointer; text-align: inherit; font-size: inherit; font-family: inherit; line-height: inherit; background: transparent; border: none; color: inherit; display: block; width: 475.203px; user-select: text;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;">tx_hash</div></button></th><th colspan="1" role="columnheader" aria-sort="none" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; text-align: inherit; font-weight: 500; position: sticky; top: 0px; height: 3rem; white-space: nowrap; background: var(--color1-200);"><button type="button" title="Toggle SortBy" style="margin: 0px; padding: 0px; box-sizing: inherit; font-weight: inherit; cursor: pointer; text-align: inherit; font-size: inherit; font-family: inherit; line-height: inherit; background: transparent; border: none; color: inherit; display: block; width: 64.8047px; user-select: text;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;">evt_index</div></button></th></tr></thead><tbody role="rowgroup" style="margin: 0px; padding: 0px; box-sizing: inherit;"><tr role="row" style="margin: 0px; padding: 0px; box-sizing: inherit;"><td role="cell" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; white-space: nowrap; border-bottom: 1px solid var(--color1-200); color: rgb(0, 0, 0); font-family: &quot;IBM Plex Mono&quot;; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;"><div style="margin: 0px; padding: 0px; box-sizing: inherit;">matched records count</div></div></td><td role="cell" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; white-space: nowrap; border-bottom: 1px solid var(--color1-200); color: rgb(0, 0, 0); font-family: &quot;IBM Plex Mono&quot;; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;">0</div></td><td role="cell" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; white-space: nowrap; border-bottom: 1px solid var(--color1-200); color: rgb(0, 0, 0); font-family: &quot;IBM Plex Mono&quot;; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;">1</div></td><td role="cell" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; white-space: nowrap; border-bottom: 1px solid var(--color1-200); color: rgb(0, 0, 0); font-family: &quot;IBM Plex Mono&quot;; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="visual_datetime__8Dkmt" style="margin: 0px; padding: 0px; box-sizing: inherit; white-space: nowrap;">2022-11-22 00:00</div></td><td role="cell" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; white-space: nowrap; border-bottom: 1px solid var(--color1-200); color: rgb(0, 0, 0); font-family: &quot;IBM Plex Mono&quot;; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;"><div style="margin: 0px; padding: 0px; box-sizing: inherit;">polygon</div></div></td><td role="cell" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; white-space: nowrap; border-bottom: 1px solid var(--color1-200); color: rgb(0, 0, 0); font-family: &quot;IBM Plex Mono&quot;; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;"><div style="margin: 0px; padding: 0px; box-sizing: inherit;">sushiswap</div></div></td><td role="cell" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; white-space: nowrap; border-bottom: 1px solid var(--color1-200); color: rgb(0, 0, 0); font-family: &quot;IBM Plex Mono&quot;; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;">1</div></td><td role="cell" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; white-space: nowrap; border-bottom: 1px solid var(--color1-200); color: rgb(0, 0, 0); font-family: &quot;IBM Plex Mono&quot;; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="" style="margin: 0px; padding: 0px; box-sizing: inherit;"><div style="margin: 0px; padding: 0px; box-sizing: inherit;">0x65c15eccd09b70a984d39c61acd4d3e7b5dfa461ac761ccfdc87bae92077391f</div></div></td><td role="cell" style="margin: 0px; padding: 0.5rem 1rem; box-sizing: inherit; white-space: nowrap; border-bottom: 1px solid var(--color1-200); color: rgb(0, 0, 0); font-family: &quot;IBM Plex Mono&quot;; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><br class="Apple-interchange-newline">

Comment (Hosuke) [reply seeds/dex/trades/dex_trades_seed.csv:157]: https://dune.com/queries/1908357
This is an unmatched seed.
Polygonscan: https://polygonscan.com/tx/0x65c15eccd09b70a984d39c61acd4d3e7b5dfa461ac761ccfdc87bae92077391f

(It seems this record is not shown in the model query result, but it should be a sushiswap polygon trade record)

File: seeds/dex/trades/dex_trades_seed.csv [hunk 4]
```diff
@@ -146,6 +146,15 @@ bnb,biswap,1,2022-11-22,0x9050a1e2f7aa392e39195b5b22382d595e94104b72699655b198c4
...
 bnb,biswap,1,2022-11-22,0xaf03946d9b98e5d6e841bb50555208801e48d773a8dfdf1ecc203d943c27c603,350,0x2170ed0880ac9a755fd29b2688956bd959f933f8,6.247796965743473757,0x7130d2a12b9bcbfae4f2634d864a1ee1ce3ead9c,0.43849
+bnb,sushiswap,1,2022-11-22,0x683b4299466b4fec3fac2859e2dd27215f69879b31a7384593b46aef71952ed5,373,0xa0ed3c520dc0632657ad2eaaf19e26c4fd431a84,6571.818499746365907562,0xe9e7cea3dedca5984780bafc599bd69add087d56,20
+bnb,sushiswap,1,2022-11-22,0xa572f3ea4bcc491c3d6a3b54f2cd59c776238c3187c1ee7975d9635f8a55d129,260,0xe9e7cea3dedca5984780bafc599bd69add087d56,30.330984668706979348,0xa0ed3c520dc0632657ad2eaaf19e26c4fd431a84,10000
+bnb,sushiswap,1,2022-11-22,0xfac7881f1461c9c56a11f68647385ea27c4d66849f646d5f377e558014652df2,536,0xe9e7cea3dedca5984780bafc599bd69add087d56,0.003061259047161221,0xa0ed3c520dc0632657ad2eaaf19e26c4fd431a84,1
+bnb,sushiswap,1,2023-01-17,0x5169181339080e2d56c13c0a075dac19f19e3f8300a8c068e8f4519bf424e7ab,186,0xe9e7cea3dedca5984780bafc599bd69add087d56,2.487085655648097925,0xc598275452fa319d75ee5f176fd3b8384925b425,177.678
```
Comment (Hosuke) [inline seeds/dex/trades/dex_trades_seed.csv:152]: test_description | result (model) | expected (seed) | block_date | blockchain | project | version | tx_hash | evt_index
-- | -- | -- | -- | -- | -- | -- | -- | --
matched records count | 0 | 1 | 2022-11-22 00:00 | bnb | sushiswap | 1 | 0x683b4299466b4fec3fac2859e2dd27215f69879b31a7384593b46aef71952ed5 | 373
matched records count | 0 | 1 | 2022-11-22 00:00 | bnb | sushiswap | 1 | 0xa572f3ea4bcc491c3d6a3b54f2cd59c776238c3187c1ee7975d9635f8a55d129 | 260
matched records count | 0 | 1 | 2022-11-22 00:00 | bnb | sushiswap | 1 | 0xfac7881f1461c9c56a11f68647385ea27c4d66849f646d5f377e558014652df2 | 536
matched records count | 0 | 1 | 2023-01-17 00:00 | bnb | sushiswap | 1 | 0x5169181339080e2d56c13c0a075dac19f19e3f8300a8c068e8f4519bf424e7ab | 186

Comment (Hosuke) [reply seeds/dex/trades/dex_trades_seed.csv:152]: These 4 records are not shown in the model query results.
https://dune.com/queries/1908356

**Review by Hosuke — CHANGES_REQUESTED** [review]
There are unmatched records in the seed files. Some trades are not shown in the models query results.

(Some trades may not be recorded in the evt_Swap source table, we may need to add more sources)

**Comment by developerfred** [comment]
Thanks for the review, I will check this seed again tomorrow morning. I’ll get some current transactions

**Comment by Hosuke** [comment]
> Thanks for the review, I will check this seed again tomorrow morning. I’ll get some current transactions

Please keep the seeds(I think they are legal), I think there are other source tables need to be included in the model.
ie.
[  sushi_bnb.UniswapV2Router02_call_swapETHForExactTokens
    , sushi_bnb.UniswapV2Router02_call_swapExactETHForTokens
    , sushi_bnb.UniswapV2Router02_call_swapExactETHForTokensSupportingFeeOnTransferTokens
    , ... ]

File: seeds/dex/trades/dex_trades_seed.csv (hunk 4 above)
Comment (Hosuke) [reply seeds/dex/trades/dex_trades_seed.csv:152]: I think these trades are by `sushi_bnb.UniswapV2Router02`, please have a look in Dune V2 decoded tables.

**Comment by developerfred** [comment]
@Hosuke added new tables to sushiswap polygon and bnb module

File: models/sushiswap/bnb/sushiswap_bnb_trades.sql [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        ON s1.contract_address = t.contract_address 
+    INNER JOIN {{ source('sushi_bnb', 'swapTokensForExactETH') }} s2
+        ON s2.contract_address = t.contract_address 
+    INNER JOIN {{ source('sushi_bnb', 'swapExactETHForTokensSupportingFeeOnTransferTokens') }} s3
+        ON s3.contract_address = t.contract_address
+        AND s3.call_block_number = s1.call_block_number
+        AND s3.call_tx_hash = s1.call_tx_hash
+        AND s3.amountETHMin = s1.amountOut
+    INNER JOIN {{ source('sushi_bnb', 'swapETHForExactTokens') }} s4
+        ON s4.contract_address = t.contract_address
```
Comment (Hosuke) [inline models/sushiswap/bnb/sushiswap_bnb_trades.sql:80]: These trade evt tables for `Router02` are actually need to be `UNION ALL`, not `INNER JOIN`.

**Review by Hosuke — CHANGES_REQUESTED** [review]
I will try to write a basic/demo version for the UNION ALL parts in `sushi_bnb_trades`, meanwhile you can fix the sources.yml based on [dbt-test prompts](https://github.com/duneanalytics/spellbook/actions/runs/3972889370/jobs/6811177763)

**Comment by Hosuke** [comment]
Hi @developerfred, the implementation [sushiswap_bnb_router02_trades.sql](https://github.com/duneanalytics/spellbook/pull/2000/files#diff-8814cd0845b75896c701861a9c073005f941b61052a7272e5c7ae227e9629633) is not finished yet(close, but need to review). But by design this router02_trades should be able to cover the seeds you added for sushiswap_bnb_trades.

File: models/sushiswap/bnb/sushiswap_bnb_trades.sql (hunk 5 above)
Comment (developerfred) [reply models/sushiswap/bnb/sushiswap_bnb_trades.sql]: done

File: models/sushiswap/polygon/sushiswap_polygon_trades.sql [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+    FROM {{ source('sushi_polygon', 'UniswapV2Pair_evt_Swap') }} t
+    SELECT * FROM {{ source('sushi_polygon', 'UniswapV2Router02_call_swapETHForExactTokens') }} s1
+    UNION ALL
+    SELECT * FROM {{ source('sushi_polygon', 'UniswapV2Router02_call_swapExactETHForTokens') }} s2
+    UNION ALL
+    SELECT * FROM {{ source('sushi_polygon', 'UniswapV2Router02_call_swapExactETHForTokensSupportingFeeOnTransferTokens') }} s3
+    UNION ALL
+    SELECT * FROM {{ source('sushi_polygon', 'UniswapV2Router02_call_swapExactTokensForETH') }} s4
+    SELECT * FROM {{ source('sushi_polygon', 'UniswapV2Router02_call_swapExactTokensForETHSupportingFeeOnTransferTokens') }} s5
```
Comment (Hosuke) [inline models/sushiswap/polygon/sushiswap_polygon_trades.sql:65]: Since you can only have one select in one CTE, you may need to fix this CTE.

**Review by Hosuke — CHANGES_REQUESTED** [review]
Hi @developerfred, this PR is getting a bit confusing.
Do you want to separate `polygon_trades` into a different PR?

Or we can learn from the [dbt-test info](https://github.com/duneanalytics/spellbook/actions/runs/3987310718/jobs/6836960567#step:9:23) to fix the code one by one?
image

We can click `Details` here for detailed dbt-test info.

File: seeds/dex/trades/dex_trades_seed.csv [hunk 7]
```diff
@@ -158,6 +158,15 @@ bnb,biswap,1,2022-11-22,0x9050a1e2f7aa392e39195b5b22382d595e94104b72699655b198c4
...
```
Comment (Hosuke) [inline seeds/dex/trades/dex_trades_seed.csv:164]: After adding router02 trades, there are still one seed for bnb is not shown in the spell
0x5169181339080e2d56c13c0a075dac19f19e3f8300a8c068e8f4519bf424e7ab
- https://bscscan.com/tx/0x5169181339080e2d56c13c0a075dac19f19e3f8300a8c068e8f4519bf424e7ab
- https://dune.com/queries/1948004?d=1

File: seeds/dex/trades/dex_trades_seed.csv [hunk 8]
```diff
@@ -158,6 +158,15 @@ bnb,biswap,1,2022-11-22,0x9050a1e2f7aa392e39195b5b22382d595e94104b72699655b198c4
...
+polygon,sushiswap,1,2022-11-22,0xcd4fda9b8492b26fb927a37f2276066d4e845d5831898c6aa3bc52195e64be29,51,0x4c3bf0a3de9524af68327d1d2558a3b70d17d42a,10,0x2791bca1f2de4661ed88a30c99a7a9449aa84174,17.471735
+polygon,sushiswap,1,2022-11-22,0xf993683656e9d580099b359bf697a15d2c704bac55a5341df59d266e00bd2214,95,0x2b9e7ccdf0f4e5b24757c1e1a80e311e34cb10c7,10,0x2791bca1f2de4661ed88a30c99a7a9449aa84174,23.507209
+polygon,sushiswap,1,2022-11-22,0x581a3ed5905f52623813259451c9bba0fb4a1b902bd2c84a04bf26831384221c,33,0xd125443f38a69d776177c2b9c041f462936f8218,160.621170255560769709,0x7ceb23fd6bc0add59e62ac25578270cff1b9f619,0.041137
+polygon,sushiswap,1,2022-11-22,0xb3fd8d3ea108e9bb55221d12e9c3d8a315e5eb1c7aad884817cdfe1d10ac0d35,38,0xd125443f38a69d776177c2b9c041f462936f8218,61.238323253823349251,0x960d43be128585ca45365cd74a7773b9d814dfbe,33.502071
+polygon,sushiswap,1,2022-11-22,0x65c15eccd09b70a984d39c61acd4d3e7b5dfa461ac761ccfdc87bae92077391f,42,0xd125443f38a69d776177c2b9c041f462936f8218,26.343507111310067208,0x60ed6acef3a96f8cdaf0c0d207bbafa66e751af2,0.087984
```
Comment (Hosuke) [inline seeds/dex/trades/dex_trades_seed.csv:169]: This seed is still not shown in the current model.
0x65c15eccd09b70a984d39c61acd4d3e7b5dfa461ac761ccfdc87bae92077391f
- https://polygonscan.com/tx/0x65c15eccd09b70a984d39c61acd4d3e7b5dfa461ac761ccfdc87bae92077391f
- https://dune.com/queries/1948014

**Review by Hosuke — COMMENTED** [review]
I will try to see if I can get missed seeds shown in the model from decoded sources.

File: models/sushiswap/bnb/sushiswap_bnb_sources.yml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+      - name: UniswapV2Router02_call_swapETHForExactTokens
+      - name: UniswapV2Router02_call_swapExactETHForTokens
+      - name: UniswapV2Router02_call_swapExactETHForTokensSupportingFeeOnTransferTokens
+      - name: UniswapV2Router02_call_swapExactTokensForETH
+      - name: UniswapV2Router02_call_swapExactTokensForETHSupportingFeeOnTransferTokens
+      - name: UniswapV2Router02_call_swapExactTokensForTokens
+      - name: UniswapV2Router02_call_swapExactTokensForTokensSupportingFeeOnTransferTokens
+      - name: UniswapV2Router02_call_swapTokensForExactETH
+      - name: UniswapV2Router02_call_swapTokensForExactTokens
+      - name: UniswapV2Pair_evt_Swap
```
Comment (Hosuke) [inline models/sushiswap/bnb/sushiswap_bnb_sources.yml:18]: cc. @jeff-dude 
There may be multiple addresses/instances as `UniswapV2Pair` on chain.
We may need to decode and `union all` all these contracts' evts at the same time.

Comment (developerfred) [reply models/sushiswap/bnb/sushiswap_bnb_sources.yml:18]: cool, i'll work on it this weekend

Comment (jeff-dude) [reply models/sushiswap/bnb/sushiswap_bnb_sources.yml:18]: feel free to tag me next week if you guys would like me to look 👍  thanks for the patience and continual builds!

**Comment by Hosuke** [comment]
I think I will remove/save the extra unfinished work in this PR, and make this PR only having a most simple version(MVP) for sushiswap binance trades to merge first.

**Comment by jeff-dude** [comment]
running through old PRs, closing due to inactivity

**Pull request closed without merging**